supplementary/alica_codegeneration/alica-plan-designer-fx-codegeneration/src/main/java/de/unikassel/vs/alica/generator/cpp/XtendTemplates.xtend

```diff
@@ -1121,4 +1121,86 @@ std::shared_ptr<UtilityFunction> UtilityFunction«plan.id»::getUtilityFunction(
     /*PROTECTED REGION END*/
 }
 '''
+
+def String transitionConditionHeader(List<Condition> conditions) '''
+#pragma once
+
+namespace alica
+{
+«FOR condition : conditions»
+static bool condition«condition.getName()»«condition.getId()»(const Blackboard* input, const RunningPlan* rp, const IAlicaWorldModel* wm);
+«ENDFOR»
+} /* namespace alica */
+'''
+
+def String transitionConditionSource(List<Condition> conditions, String pkgName) '''
+«IF (conditions.get(0).relativeDirectory == null || conditions.get(0).relativeDirectory.isEmpty)»
+#include <«pkgName»/condition.h>
+«ELSE»
+#include <«pkgName»/condition.relativeDirectory».h>
+«ENDIF»
+
+namespace alica
+{
+«FOR condition : conditions»
+static bool condition«condition.getName()»«condition.getId()»(const Blackboard* input, const RunningPlan* rp, const IAlicaWorldModel* wm)
+{
+/*PROTECTED REGION ID(condition«condition.id») ENABLED START*/
+        «IF (protectedRegions.containsKey("condition" + condition.id))»
+«protectedRegions.get("condition" + condition.id)»
+        «ELSE»
+            std::cout << "The condition «condition.id» is not implement yet!" << std::endl;
```

Comment:
can add a `static_assert(false, "Condition <condition_id> with name <condition_name> is not yet implemented)` so that the error is caught during compilation itself